tests/handlers/test_user.py

```diff
@@ -111,6 +114,57 @@ def test_post_user_as_regular_user(test_client, db_session, user,
     assert response.json()['detail'] == "Permission denied."
 
 
+def test_update_password_successful(test_client, db_session, session_cookie,
+                                    user):
+    response = test_client.put(
+        '/user/ccaf8799-b134-4e47-82f1-a4d9a207c040',
+        json={'password': 'newpassword', 'password_confirm': 'newpassword',
+              'old_password': 'testing123'},
+        cookies=session_cookie
+    )
+
+    assert response.status_code == 200
+    user = db_session.query(User).get(user.id)
+    assert compare_plaintext_to_hash(
+        'newpassword', user.password_hash, user.password_salt) is True
+
+
+def test_update_password_wrong_password(test_client, db_session,
```

Comment:
No test to assert functionality on a `password` `password_confirm` mismatch,

Reply:
But i did that in the schema right?

Reply:
In that case you'd see some kind of test to assert whether that schema is guaranteed to be used in this handler, which I wouldn't recommend. We can't stop asserting things because we know that currently we use a piece of code that's asserted at a lower level.

I can remove the schema code from the handler in question and replace it with something that omits the `password == password_confirm` check and all tests would succeed.

Reply:
Alright